### PR TITLE
fix(mysql): route routine DDL through text protocol

### LINKED ISSUE #347: fix(mysql): route routine DDL through text protocol
## Bug Description

Executing MySQL routine DDL through the MCP can fail because the built-in MySQL driver sends non-result statements through the prepared statement protocol.

Reported error:

```text
error returned from database: 1295 (HY000): This command is not supported in the prepared statement protocol yet
DROP PROCEDURE IF EXISTS sociedades_close;
```

## Steps to Reproduce

1. Configure a MySQL connection in Tabularis.
2. Execute through the MCP:
   ```sql
   DROP PROCEDURE IF EXISTS sociedades_close;
   ```
3. Observe MySQL error 1295 from the prepared statement protocol.

## Expected Behavior

Routine DDL such as DROP/CREATE/ALTER PROCEDURE and FUNCTION should execute successfully when MySQL requires text protocol.

## Actual Behavior

The statement is routed through the prepared statement protocol and MySQL rejects it with error 1295.

## Scope

- MySQL driver only.
- MCP benefits automatically because it uses the normal driver execute_query path.

## Verification

Focused classifier tests pass with:

```bash
cargo test text_protocol --lib
```

### PR DESCRIPTION
**Pull request by Stiwar0098**
Closes #347

## PR Type

- [x] Bug fix

## Summary

- Route MySQL routine DDL through text protocol instead of prepared statements.
- Cover PROCEDURE/FUNCTION create, alter, drop statements, including CREATE DEFINER variants.
- Ignore local `.atl/` and `.opencode/` tooling folders.

## Changes

| File | Change |
|------|--------|
| `.gitignore` | Ignore local AI/tooling directories. |
| `src-tauri/src/drivers/mysql/mod.rs` | Extend MySQL text-protocol routing for routine DDL unsupported by prepared statements. |
| `src-tauri/src/drivers/mysql/tests.rs` | Add classifier coverage for routine DDL and negative cases. |

## Test Plan

- [x] `cargo test text_protocol --lib`
- [x] Focused diff review completed before PR.
- [ ] Live MCP + MySQL execution not run locally.

## Notes

MySQL documentation lists the SQL statements permitted in prepared statements and routine DDL is not included. The reported `DROP PROCEDURE IF EXISTS sociedades_close;` now routes through `sqlx::raw_sql()` via the existing MySQL text-protocol path.

### CODE REVIEW

**Review by Stiwar0098 — COMMENTED** [review]
Addressed the Kilo review feedback.

This update routes CREATE OR REPLACE PROCEDURE and CREATE OR REPLACE FUNCTION through the MySQL text-protocol path, including DEFINER variants. I also tightened the classifier so CREATE [OR REPLACE] DEFINER ... VIEW statements do not match just because their body mentions PROCEDURE or FUNCTION.

Verification: cargo test text_protocol --lib passed via the Tabularis build-skill Rust test path, 5 passed, 0 failed.

File: src-tauri/src/drivers/mysql/mod.rs [hunk 1]
```diff
@@ -892,6 +897,13 @@ fn is_text_protocol_stmt(query: &str) -> bool {
         || head.starts_with("RELEASE SAVEPOINT")
         || head.starts_with("LOCK TABLES")
         || head.starts_with("UNLOCK TABLES")
+        || head.starts_with("DROP PROCEDURE")
+        || head.starts_with("CREATE PROCEDURE")
```
[the remark this answers is not in the document]
Comment (Stiwar0098) [reply src-tauri/src/drivers/mysql/mod.rs]: Fixed in commit 3ee7d39. The classifier now routes CREATE OR REPLACE PROCEDURE and CREATE OR REPLACE FUNCTION through text protocol, including compact and spaced DEFINER variants.

I also tightened the matching so CREATE [OR REPLACE] DEFINER ... VIEW does not route through text protocol just because the view body mentions PROCEDURE or FUNCTION.

Verification: cargo test text_protocol --lib passed via the Tabularis build-skill Rust test path, 5 passed, 0 failed.

[the remark this answers is not in the document]
Comment (Stiwar0098) [reply src-tauri/src/drivers/mysql/mod.rs]: Fixed in commit 3ee7d39. This duplicate finding is covered by the same update: CREATE OR REPLACE routine DDL now uses the text-protocol path for PROCEDURE/FUNCTION statements, with tests for direct, compact DEFINER, and spaced DEFINER = 'root' @ 'localhost' forms.

Verification: cargo test text_protocol --lib passed via the Tabularis build-skill Rust test path, 5 passed, 0 failed.

**Comment by debba** [comment]
@Stiwar0098 Looks good overall — the DEFINER/VIEW-body handling is the tricky part and it's done right. A few nits:

1. The new parsing helpers (`after_definer_clause`, `find_first_top_level_object_keyword`, `is_keyword_boundary`, `definer_stmt_is_routine`) live in `mod.rs`. Per our Rust rules pure parsing logic should sit in a sibling module — could pull these into something like `mysql/stmt_classify.rs` and keep `mod.rs` to orchestration.

2. The tests only hit `is_text_protocol_stmt`, so the helpers are covered indirectly. Coverage is effectively complete (the spaced-definer and quoted-body cases are exactly the helper's edge cases), but a couple of direct `#[test]`s on `find_first_top_level_object_keyword` would be nice to have.

3. `starts_with("CREATE DEFINER")` / `"CREATE OR REPLACE DEFINER"` assume single spaces, so `CREATE  DEFINER` or `CREATE OR  REPLACE` would slip through. I'd handle this one — collapsing runs of whitespace on the head before matching (or matching token-wise) would close it, and it's cheap. Worth a test case too.

Let's get 1/2/3 in and then it's good to merge.
